sdk/java/pulumi/src/main/java/com/pulumi/provider/internal/Provider.java

```diff
@@ -0,0 +1,11 @@
+package com.pulumi.provider.internal;
+
+import java.util.concurrent.CompletableFuture;
+
+import com.pulumi.provider.internal.models.*;
+
+public abstract class Provider {
+    public CompletableFuture<GetSchemaResponse> getSchema(GetSchemaRequest request) {
+        throw new UnsupportedOperationException("Method 'getSchema' is not implemented");
+    }
+}
```

Comment:
Part of me thinks we should prefer interfaces with default implementations to abstract classes, but I don't know if there are important reasons to/to not to, so happy either way. I think `Provider` was an interface in TS and we found that a bit problematic at times, but TS interfaces don't support default implementations, so maybe that was it. Not sure.

Reply:
It's all internal for now in any case.

Reply:
At the point of this PR, interface sounds like a great idea. I converted it for now, let's see how we want it to evolve later. It's all internal for now, indeed.